File: 0875-koko-eating-bananas/0875-koko-eating-bananas.java
class Solution {

    boolean possible(int bananas,int h,int piles[]){
        int timetaken=0;
        for(int x:piles){
            timetaken+=(x+bananas-1)/bananas;
            if(timetaken>h) return false;
        }
        return true;
    }

    public int minEatingSpeed(int[] piles, int h) {
        int low=1,high=Arrays.stream(piles).max().getAsInt();
        while(low<=high){
            int mid=low+(high-low)/2;
            if(possible(mid,h,piles)) high=mid-1;
            else low=mid+1;
        }
        return low;
    }
}

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cs
public class Solution {

    public bool possible(int mid,int[] piles,int h){
        int time=0;
        foreach(int x in piles){
            time+=(x+mid-1)/mid;
            if(time>h) return false;
        }
        return true;
    }

    public int MinEatingSpeed(int[] piles, int h) {
        int n=piles.Length;
        int low=1,high=piles.Max();
        while(low<=high){
            int mid=low+(high-low)/2;
            if(possible(mid,piles,h)) high=mid-1;
            else low=mid+1;
        }
        return low;
    }
}